DTML.EduBot/Common/MessageTranslator.cs

```diff
@@ -0,0 +1,107 @@
+{
+    public static class MessageTranslator
+    {
+        /// <summary>
+        /// Microsoft translator API key
+        /// </summary>
+        private static string TranslatorApiKey = "";
+
+        /// <summary>
+        /// Language to which we want the input to be converted to
+        /// </summary>
+        private static string inputTextLang;
+
+        /// <summary>
+        /// Translate text to english and send forward
+        /// </summary>
+        /// <param name="inputText">input text to be converted</param>
+        /// <param name="inputLang">Language to which the inputText should be translated to </param>
+        /// <returns>translated string</returns>
+        public static async Task<string> TranslateTextAsync(string inputText,  string inputLang = null) 
+        {
+            var accessToken = await MessageTranslator.GetAuthenticationTokenAsync(TranslatorApiKey);
+            if (inputLang == null)
+            {
+                inputLang = inputTextLang;
+            }
+
+            string query = $"?text={System.Net.WebUtility.UrlEncode(inputText)}&to={inputLang}&contentType=text/plain";
+
+            using (var client = new HttpClient())
+            {
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", accessToken);
+                var response = await client.GetAsync(System.Configuration.ConfigurationManager.AppSettings["translatorEndpoint"] + query);
+                var result = await response.Content.ReadAsStringAsync();
+
+                if (!response.IsSuccessStatusCode)
+                    return null;
+
+                if (result != null)
+                {
+                    var translatedText = XElement.Parse(result).Value;
+                    return translatedText;
+                }
+
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Identify language for input text
+        /// </summary>
+        /// <param name="inputText">text whose language should be detected</param>
+        /// <returns></returns>
+        public static async Task IdentifyLangAsync(string inputText)
+        {
+            var accessToken = await MessageTranslator.GetAuthenticationTokenAsync(TranslatorApiKey);
+            string query = $"?text={System.Net.WebUtility.UrlEncode(inputText)}&contentType=text/plain";
+
+            using (var client = new HttpClient())
+            {
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", accessToken);
+                var response = await client.GetAsync(System.Configuration.ConfigurationManager.AppSettings["detectLanguageEndpoint"] + query);
+                var result = await response.Content.ReadAsStringAsync();
+
+                if (!response.IsSuccessStatusCode)
+                    return;
+
+                if (result != null)
+                {
+                    var translatedText = XElement.Parse(result).Value;
```

Comment:
This is fine. 

Just a suggestion: WCF services can be added as service reference, so parsing this is handled by the framework. So we directly can call the DetectLanguage method. If you have access to DTML code, that would be an example. Anyway, good work 👍